lib/ecto/changeset.ex

```diff
@@ -733,6 +743,17 @@ defmodule Ecto.Changeset do
       struct = model.__struct__
       query  = from m in struct, select: field(m, ^field), limit: 1
 
+      if opts[:scope] do
+        query = Enum.reduce(opts[:scope], query, fn(field, acc) ->
+          value = get_field(changeset, field)
+          if value do
```

Comment:
Oh, there are two cases here

`validate_unique(changeset, :title, scope: [:field_that_doesnt_exist], on: Repo)`

and

`validate_unique(changeset, :title, scope: [:field_that_exists_but_is_nil], on: Repo)`